msal/src/request/java/com/microsoft/identity/client/InteractiveRequest.java

```diff
@@ -127,6 +140,73 @@ synchronized void preTokenRequest() throws MsalUserCancelException, MsalClientEx
         processAuthorizationResult(sAuthorizationResult);
     }
 
+    /**
+     * Create the {@Link MicrosoftStsAuthorizationRequest} from authentication parameters.
+     *
+     * @return MicrosoftStsAuthorizationRequest
+     * @throws MsalClientException
+     */
+    private MicrosoftStsAuthorizationRequest createAuthRequest() throws MsalClientException {
+        MicrosoftStsPromptBehavior promptBehavior;
+
+        //Map the UIBehavior from MSAL to Common
+        switch (getAuthRequestParameters().getUiBehavior()) {
+            case CONSENT:
+                promptBehavior = MicrosoftStsPromptBehavior.CONSENT;
+                break;
+            case FORCE_LOGIN:
+                promptBehavior = MicrosoftStsPromptBehavior.FORCE_LOGIN;
+                break;
+            default:
+                promptBehavior = MicrosoftStsPromptBehavior.SELECT_ACCOUNT;
+                break;
+        }
+
+        final MicrosoftStsAuthorizationRequest authorizationRequest = new MicrosoftStsAuthorizationRequest(
+                OauthConstants.Oauth2Parameters.CODE,
+                getAuthRequestParameters().getClientId(),
+                getAuthRequestParameters().getRedirectUri(),
+                null,
+                getAuthRequestParameters().getScope(),
+                getAuthRequestParameters().getAuthority().getAuthorityUrl(),
+                getAuthRequestParameters().getAuthority().getAuthorizeEndpoint(),
+                getAuthRequestParameters().getLoginHint(),
+                getAuthRequestParameters().getRequestContext().getCorrelationId(),
+                null,
+                getAuthRequestParameters().getExtraQueryParam(),
+                PublicClientApplication.getSdkVersion(),
+                promptBehavior,
+                null,
+                null,
+                null,
+                getAuthRequestParameters().getSliceParameters(),
+                null);
+
+        if (null != getAuthRequestParameters().getUser()) {
+            authorizationRequest.setUid(getAuthRequestParameters().getUser().getUid());
```

Comment:
I'd prefer if this method weren't here and if this class didn't know what type of `AuthorizationRequest` it is handling BUT I have a minor nit about this code block too: we derefence `getAuthRequestParameters().getUser()` 3 times.

For wherever this method ends-up, can we get the `User` once inside this scope and then call the getters on it?
```java
final User user = getAuthRequestParameters().getUser();
authorizationRequest.setUid(user.getUid());
authorizationRequest.setUtid(user.getUtid());
authorizationRequest.setDisplayableId(user.getDisplayableId());
```